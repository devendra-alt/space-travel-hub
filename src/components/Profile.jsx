import React from 'react';
import {
  Alert, Button, Col, Container, ListGroup, Row,
} from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import Header from './Header';
import { rocketBookingCancel } from '../redux/rockets/features/rocketsSlice';

function Profile() {
  const dispath = useDispatch();
  const { rockets } = useSelector((state) => state.rockets);
  const reservedRockets = rockets.filter((rocket) => rocket.reserved === true);

  const { missions } = useSelector((state) => state.missions);
  const reservedMissions = missions.filter((mission) => mission.reserved);

  return (
    <>
      <Header />
      <Container fluid className="mx-auto col-10">
        <Row className="flex-column flex-lg-row flex-xl-row">
          <Col>
            <div className="mb-4">
              <p className="h2">Missions</p>
              <ListGroup>
                {/* Render reserved missions here */}
                {reservedMissions.map((mission) => (
                  <ListGroup.Item key={mission.mission_id}>
                    <p style={{ fontSize: '1.4rem' }}>{mission.mission_name}</p>
                  </ListGroup.Item>
                ))}
              </ListGroup>
            </div>
          </Col>
          <Col>
            {!reservedRockets.length && (
              <Alert variant="info">My Reserved Rockets is empty</Alert>
            )}
            {!!reservedRockets.length && (
              <ListGroup>
                <p className="h2">My Rockets</p>
                {reservedRockets.map((rocket) => (
                  <ListGroup.Item
                    key={rocket.id}
                    className="d-flex flex-sm-column flex-md-row flex-lg-row justify-content-between align-items-center"
                  >
                    <div>
                      <p style={{ fontSize: '1.4rem' }}>{rocket.name}</p>
                      <Button variant="outline-info">
                        <a href={rocket.wikipedia} target="_blank" rel="noreferrer">
                          Read more
                        </a>
                      </Button>
                    </div>
                    <Button
                      variant="outline-danger"
                      onClick={() => dispath(rocketBookingCancel(rocket.id))}
                    >
                      Cancel Reservation
                    </Button>
                  </ListGroup.Item>
                ))}
              </ListGroup>
            )}
          </Col>
        </Row>
      </Container>
    </>
  );
}

export default Profile;
