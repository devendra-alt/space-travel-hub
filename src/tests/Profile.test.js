import '@testing-library/jest-dom';
import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Profile from '../components/Profile';

// Mock Redux store
const mockStore = configureStore([]);
const initialState = {
  rockets: {
    rockets: [
      {
        id: '1',
        name: 'Falcon 9',
        reserved: true,
        wikipedia: 'https://en.wikipedia.org/wiki/Falcon_9',
      },
    ],
  },
  missions: {
    missions: [
      {
        mission_id: '1',
        mission_name: 'Mock Mission 1',
        reserved: true,
      },
    ],
  },
};
const store = mockStore(initialState);

test('Profile component renders correctly', () => {
  const { asFragment } = render(
    <Provider store={store}>
      <BrowserRouter>
        <Profile />
      </BrowserRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
