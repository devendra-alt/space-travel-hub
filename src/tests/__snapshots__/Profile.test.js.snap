// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile component renders correctly 1`] = `
<DocumentFragment>
  <header
    class="main-header"
  >
    <div
      class="site-title-section"
    >
      <img
        alt="space travel hub logo"
        height="60px"
        src="test-file-stub"
        width="60px"
      />
      <h1
        class="site-title"
      >
        Space Traveler's Hub
      </h1>
    </div>
    <nav
      class="main-header-nav"
    >
      <a
        aria-current="page"
        class="nav-link active"
        href="/"
      >
        Rockets
      </a>
      <a
        class="nav-link"
        href="/missions"
      >
        Missions
      </a>
      <a
        class="nav-link"
        href="/my-profile"
      >
        Profile
      </a>
    </nav>
  </header>
  <div
    class="mx-auto col-10 container-fluid"
  >
    <div
      class="flex-column flex-lg-row flex-xl-row row"
    >
      <div
        class="col"
      >
        <div
          class="mb-4"
        >
          <p
            class="h2"
          >
            Missions
          </p>
          <div
            class="list-group"
          >
            <div
              class="list-group-item"
            >
              <p
                style="font-size: 1.4rem;"
              >
                Mock Mission 1
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <div
          class="list-group"
        >
          <p
            class="h2"
          >
            My Rockets
          </p>
          <div
            class="d-flex flex-sm-column flex-md-row flex-lg-row justify-content-between align-items-center list-group-item"
          >
            <div>
              <p
                style="font-size: 1.4rem;"
              >
                Falcon 9
              </p>
              <button
                class="btn btn-outline-info"
                type="button"
              >
                <a
                  href="https://en.wikipedia.org/wiki/Falcon_9"
                  rel="noreferrer"
                  target="_blank"
                >
                  Read more
                </a>
              </button>
            </div>
            <button
              class="btn btn-outline-danger"
              type="button"
            >
              Cancel Reservation
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
